src/components/ListOfCards.js

```diff
@@ -2,24 +2,29 @@ import React, { Component } from 'react';
 import styled from 'styled-components'
 import Card from "./Card";
 import { getKudoses } from '../api/KudosApi'
+import Spinner from "./Spinner";
 
 class ListOfCards extends Component {
     state = {
-        kudoses: []
+        kudoses: [],
+        loading: true
     }
     componentWillMount () {
```

Comment:
This kind of activity should take place in `componentDidMount`. The reason is that the `then` may happen (not likely to happen very often with the real API) before component is mounted, thus trying to `setState` and further change the component which is not yet mounted.
Due to this, React Team decided to drop this lifecycle method:
https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

```
UNSAFE_componentWillMount() is invoked just before mounting occurs. It is called before render(), therefore calling setState() synchronously in this method will not trigger an extra rendering. Generally, we recommend using the constructor() instead for initializing state.

Avoid introducing any side-effects or subscriptions in this method. For those use cases, use componentDidMount() instead.
```